### ISSUE TITLE
[Sample] Add a policy for Consul Service Mesh for TLS Min - Mesh Configuration 

### ISSUE THREAD

**Issue by shivaylamba** [kyverno/policies#446]
### Problem Statement

We want to create a policy for Consul service mesh to add a policy for the mesh configuration. The policy will check the TLS Min version to ensure that whenever the mesh is configured, there is a minimum version of TLS set for all the service mesh proxies.

### Solution Description

Here is an example of a sample policy:
 ```
apiVersion: kyverno.io/v1
kind: ClusterPolicy
metadata:
  name: consul-mesh-policy
  annotations:
    policies.kyverno.io/title: Consul TLS Min
    policies.kyverno.io/category: Service Mesh Policy Check
    policies.kyverno.io/severity: medium
    policies.kyverno.io/subject: 'Service Mesh'
    policies.kyverno.io/description: >-
      This policy will check the TLS Min version to ensure that whenever the mesh is configured, there is a minimum version of TLS set for all the service mesh proxies.
spec:
  validationFailureAction: enforce
  background: true
  rules:
    - name: check-for-tls-version
      match:
        resources:
          kinds:
            - Mesh
      validate:
        message: The minimum versior of TLS is TLS v1_2
        pattern:
          spec:
              tls:
                incoming:
                  tlsMinVersion: TLSv1_2

```

### Other Comments

https://developer.hashicorp.com/consul/docs/connect/config-entries/mesh

### Slack discussion

https://kubernetes.slack.com/archives/CLGR9BJU9/p1667314129192989

### Troubleshooting

- [X] I have read and followed the documentation AND the [troubleshooting guide](https://kyverno.io/docs/troubleshooting/).
- [X] I have verified the policy does not exist in the [samples library](https://kyverno.io/policies/).

**Comment by shivaylamba** [comment]
cc @ashwinexe

**Comment by chipzoller** [comment]
Please follow the [contribution guidelines](https://github.com/kyverno/policies#contribution) and open a PR against this repository. Please add one positive (passing) and one negative (failing) test case using the [Kyverno CLI](https://kyverno.io/docs/kyverno-cli/). There are many examples in this repo of such tests which you may use to model yours.

**Comment by chipzoller** [comment]
Closed in #474
